File: src/main/java/ro/mpp2024/Domain/Participant.java
package ro.mpp2024.Domain;

import java.util.Objects;

public class Participant extends Entity<Integer>{
    private String nume;
    private String prenume;
    private Integer varsta;

    public Participant(String nume, String prenume, Integer varsta) {
        this.nume = nume;
        this.prenume = prenume;
        this.varsta = varsta;
    }

    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public String getPrenume() {
        return prenume;
    }

    public void setPrenume(String prenume) {
        this.prenume = prenume;
    }

    public Integer getVarsta() {
        return varsta;
    }

    public void setVarsta(Integer varsta) {
        this.varsta = varsta;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        Participant that = (Participant) o;
        return Objects.equals(nume, that.nume) && Objects.equals(prenume, that.prenume) && Objects.equals(varsta, that.varsta);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), nume, prenume, varsta);
    }

    @Override
    public String toString() {
        return "Participant{" +
                "nume='" + nume + '\'' +
                ", prenume='" + prenume + '\'' +
                ", varsta=" + varsta +
                '}';
    }
}

File: SwimmingCompetitionCSharpGUI/Domain/Participant.cs
namespace SwimmingCompetition.Domain;

public class Participant: Entity<int>
{
    public String Nume { get; set; }
    public String Prenume { get; set; }
    public Int32 Varsta { get; set; }

    public Participant(string nume, string prenume, int varsta)
    {
        Nume = nume;
        Prenume = prenume;
        Varsta = varsta;
    }

    protected bool Equals(Participant other)
    {
        return base.Equals(other) && Nume == other.Nume && Prenume == other.Prenume && Varsta == other.Varsta;
    }

    public override bool Equals(object? obj)
    {
        if (ReferenceEquals(null, obj)) return false;
        if (ReferenceEquals(this, obj)) return true;
        if (obj.GetType() != this.GetType()) return false;
        return Equals((Participant)obj);
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(base.GetHashCode(), Nume, Prenume, Varsta);
    }

    public override string ToString()
    {
        return $"{base.ToString()}, {nameof(Nume)}: {Nume}, {nameof(Prenume)}: {Prenume}, {nameof(Varsta)}: {Varsta}";
    }
}